tpu-client/src/tpu_client.rs

```diff
@@ -95,6 +105,66 @@ where
         self.invoke(self.tpu_client.try_send_transaction(transaction))
     }
 
+    /// Serialize and send transaction to the current and upcoming leader TPUs according to fanout
+    /// size
+    /// Attempts to send and confirm tx "tries" times
+    /// Waits for signature confirmation before returning
+    /// Returns the transaction signature
+    pub fn send_and_confirm_transaction_with_retries<T: Signers + ?Sized>(
+        &self,
+        keypairs: &T,
+        transaction: &mut Transaction,
```

Comment:
Do we really need a reference or can consume here?

Reply:
think it needs a reference because it's retrying, and resigning the tx

Reply:
got it